moped-etl/prefect/moped_test_create_flow.py

```diff
@@ -38,316 +44,304 @@
-        basename=basename,
-        load_balancer=load_balancer,
-        task_definition=task_definition,
-        target_group=target_group,
-        listeners_token=listeners,
-        cluster_token=cluster,
-    )
+    database_exists = db.database_exists(slug)
 
+    with case(database_exists, True):
 
-with Flow(
-    "Moped Test ECS Decommission",
-    # Observation! The hex key of the container is from the build context!!
-    # You can use this as a userful key to associate a state of the code and an environmental state!
-    run_config=UniversalRun(labels=["moped", hostname]),
-) as ecs_decommission:
+        running_tasks = ecs.check_count_running_ecs_tasks(slug=slug)
+        with case(running_tasks, True):
+            drained_service = drain_service(slug)
+        ready_to_drop_db = merge(drained_service, running_tasks)
 
-    basename = Parameter("basename")
+        remove_database = db.remove_database(
+            slug=slug, ready_to_drop_db=ready_to_drop_db
+        )
 
-    set_count_at_zero = ecs.set_desired_count_for_service(basename=basename, count=0)
+    ready_to_commission = merge(remove_database, database_exists)
 
-    tasks = ecs.list_tasks_for_service(basename=basename)
+    create_database = db.create_database(
+        slug=slug, upstream_tasks=[ready_to_commission]
+    )
 
-    stop_token = ecs.stop_tasks_for_service(
-        basename=basename, tasks=tasks, zero_count_token=set_count_at_zero
+    populate_database_command = db.populate_database_with_data_command(
+        slug=slug, stage=database_seed_source, upstream_tasks=[create_database]
     )
 
-    drained_service = ecs.wait_for_service_to_be_drained(
-        basename=basename, stop_token=stop_token
+    populate_database = db.populate_database_with_data_task(
+        command=populate_database_command
     )
 
-    no_listeners = ecs.remove_all_listeners(basename=basename)
+    ## Commission the API
 
-    no_target_group = ecs.remove_target_group(
-        basename=basename, no_listener_token=no_listeners
-    )
+    secret_exists = api.check_secret_exists(slug=slug)
 
-    no_service = ecs.delete_service(
-        basename=basename,
-        drained_token=drained_service,
-        no_target_group_token=no_target_group,
-    )
+    with case(secret_exists, True):
+        remove_api_config_secret_arn = api.remove_moped_api_secrets_entry(slug=slug)
 
-    no_cluster = ecs.remove_ecs_cluster(basename=basename, no_service_token=no_service)
+    ready_for_secret = merge(remove_api_config_secret_arn, secret_exists)
 
-    removed_load_balancer = ecs.remove_load_balancer(
-        basename=basename, no_cluster_token=no_cluster
+    create_api_config_secret_arn = api.create_moped_api_secrets_entry(
+        slug=slug, ready_for_secret=ready_for_secret
     )
 
-    removed_hostname = ecs.remove_route53_cname(
-        basename=basename, removed_load_balancer_token=removed_load_balancer
-    )
+    # test if api is deployed, remove if so
```

Comment:
I like removing and deploying again if the API exists because it means less code, but there is a [zappa update](https://github.com/zappa/Zappa#updates) command. I don't know if there is really any advantage (might be faster?), but I wanted to throw it out there so we have more info - not vouching to change this at all.